### PR TITLE
Adding 'st-nichs-reindeers-need-water.pod'

### LINKED ISSUE #182: St. Nich's Reindeers Need Water!
Which Perl module do you want to write about:

**Util::H2O**

Please give a short description of the article you want to write. Two to three sentences is fine:

With Santa's sleigh having to be moving at 650 miles per second while simultaneously visiting 822.6 homes with Children per _second_ [1], one may rightly assume that His reindeer are might thirsty for some high quality H2O! While a Perl module won't actually water Dunder and Blixem, there is one that comes around every so often that can make Perl more fun and productive, even for existing code! On this day of Advent, I will discuss the module, **Util::H2O**.

_References_:

1. According to the peered reviewed and legendary Internet report on, "_The Physics of Santa_"..

### PR DESCRIPTION
**Pull request by oodler577**
Resolves #182.

### CODE REVIEW

**Comment by oodler577** [comment]
redoing

File: 2022/incoming/st-nichs-reindeers-need-water.pod [hunk 1]
```diff
@@ -0,0 +1,148 @@
+Author: oodler@cpan.org
+Title:  St. Nich's Reindeers Need H2O!
```
Comment (oalders) [inline 2022/incoming/st-nichs-reindeers-need-water.pod:2]:
```suggestion
Title:  St. Nick's Reindeers Need H2O!
```

I believe "Nick's" would be the more common form here, but I could be wrong.

File: 2022/incoming/st-nichs-reindeers-need-water.pod [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+Topic: Perl OOP, C<HASH> references, accessors
+
+=encoding utf8
+
+=head1 St. Nich's Reindeers Need H2O!
+
+Reindeer are hearty creatures, and Santa's must be the heartiest.
```
Comment (oalders) [inline 2022/incoming/st-nichs-reindeers-need-water.pod:9]: I *think* this should be "hardy" and "hardiest", based on the context.

File: 2022/incoming/st-nichs-reindeers-need-water.pod [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+  #      "city": "Gwenborough",
+  #      "zipcode": "92998-3874",
+  #      "geo": {
+  #        "lat": "-37.3159",
+  #        "lng": "81.1496"
+  #      }
+  #    }
+  #  },
+  # ...
+  # ]
```
Comment (oalders) [inline 2022/incoming/st-nichs-reindeers-need-water.pod:90]: Could we make this a touch clearer? For `is( $got, expected )`, there is a hard-coded `got` and no expected, so it's not immediately clear to me what we're doing here.

File: 2022/incoming/st-nichs-reindeers-need-water.pod [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+  
+Finally, iterating over all persons in the list is greatly simplified
+by providing accessors; even for deeply nested C<HASH> references:
+  
+  print qq{lat, lng, name, twitter\n};
+  
+  foreach my $person (@$json_array_ref) {
+  
+    # -recurse creates deep accessors, e.g.,
```
Comment (oalders) [inline 2022/incoming/st-nichs-reindeers-need-water.pod:99]: This is neat! I had no idea something like this existed.

File: 2022/incoming/st-nichs-reindeers-need-water.pod [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+           $person->name,
+           $person->username;
+  }
+  
+  # print status to STDERR on succcess
+  printf STDERR qq{\nSummary:\nWeb request responded with with HTTP status: %d\n}, $response->status;
+
+=head1 Conclusion
+
+With St. Nich's list now complete with online usage profiles for all people
```
Comment (oalders) [inline 2022/incoming/st-nichs-reindeers-need-water.pod:116]:
```suggestion
With St. Nick's list now complete with online usage profiles for all people
```

File: 2022/incoming/st-nichs-reindeers-need-water.pod [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+on twitter, he now sets his mind on more jolly thoughts; not sugar plums and
```
Comment (oalders) [inline 2022/incoming/st-nichs-reindeers-need-water.pod:117]:
```suggestion
on Twitter, he now sets his mind on more jolly thoughts; not sugar plums and
```

File: 2022/incoming/st-nichs-reindeers-need-water.pod [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+
+=item L<Getopt::Long> (with L<Util::H2O::More>'s C<opt2h2o>)
+
+=back
+
+He may even check out using the C<bless> replacement, C<baptise>, which is
+contained in L<Util::H2O::More> and is like C<bless>, but gives you the ability
+to create accessors, too!
+
+Don't get on Santas I<Naughty List> by using cumbersome and chunky Perl I<OOP>
```
Comment (oalders) [inline 2022/incoming/st-nichs-reindeers-need-water.pod:143]:
```suggestion
Don't get on Santa's I<Naughty List> by using cumbersome and chunky Perl I<OOP>
```

File: 2022/incoming/st-nichs-reindeers-need-water.pod [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+frameworks when all you really need is a little H2O. Go checkout L<Util::H2O> and
```
Comment (oalders) [inline 2022/incoming/st-nichs-reindeers-need-water.pod:144]:
```suggestion
frameworks when all you really need is a little H2O. Go check out L<Util::H2O> and
```

File: 2022/incoming/st-nichs-reindeers-need-water.pod (hunk 2 above)
Comment (oodler577) [reply 2022/incoming/st-nichs-reindeers-need-water.pod:9]: I think I meant both but will side with _hearty_.

File: 2022/incoming/st-nichs-reindeers-need-water.pod (hunk 3 above)
Comment (oodler577) [reply 2022/incoming/st-nichs-reindeers-need-water.pod:90]: I am not sure if I removed some `got` or you're referring to something else. I'll make this section more clear, but wanted to see if you could clarify your comments.

**Pull request closed without merging**